### PR TITLE
Add doc for Iceberg source

### LINKED ISSUE #1888: Document: feat(batch): support batch read iceberg source
This issue tracks the documentation update needed for the merged PR #15214.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/15214
Source PR Merged At: 2024-02-27T09:02:20Z

### LINKED ISSUE #1915: Document: feat(frontend): derive columns from iceberg source automatically
This issue tracks the documentation update needed for the merged PR #15415.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/15415
Source PR Merged At: 2024-03-04T10:13:50Z

### PR DESCRIPTION
**Pull request by WanYixian**
## Info

- **Description**

  - Add a dedicated doc of Iceberg source
  - Support batch read Iceberg source
  - Support deriving columns from Iceberg source automatically

- **Notes**

  - BTW, reorganize the CDC docs and put them together

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/15214
  - https://github.com/risingwavelabs/risingwave/pull/15415

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1888
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1915

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/ingest/ingest-from-iceberg.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+Iceberg supports these types of catalogs:
+
+- Storage catalog: The Storage catalog stores all metadata in the underlying file system, such as Hadoop or S3. Currently, we only support S3 as the underlying file system.
+
+- REST catalog: RisingWave supports the [REST catalog](https://iceberg.apache.org/concepts/catalog/#decoupling-using-the-rest-catalog), which acts as a proxy to other catalogs like Hive, JDBC, and Nessie catalog. This is the recommended approach to use RisingWave with Iceberg tables.
+
+- Hive catalog: RisingWave supports the Hive catalog. You need to set `catalog.type` to `hive` to use it. See the full example in this [configuration file](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/hive/config.ini).
+
+- Jdbc Catalog: RisingWave supports the [JDBC catalog](https://iceberg.apache.org/docs/latest/jdbc/#configurations). See the full example in this [configuration file](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/jdbc/config.ini).
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-iceberg.md]:
```suggestion
- Jdbc catalog: RisingWave supports the [JDBC catalog](https://iceberg.apache.org/docs/latest/jdbc/#configurations). See the full example in this [configuration file](https://github.com/risingwavelabs/risingwave/blob/main/integration_tests/iceberg-sink2/docker/jdbc/config.ini).
```

Comment (ShanlanLi) [reply docs/ingest/ingest-from-iceberg.md]: You didn't capitalize other "c". So maybe this doesn't need either?

File: docs/ingest/ingest-from-iceberg.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    catalog.name='demo',
+    database.name='dev',
+    table.name='table'
+);
+```
+
+Then, you can query the Iceberg source:
+
+```sql
+select * from iceberg_source
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-iceberg.md]:
```suggestion
SELECT * FROM iceberg_source;
```

**Review by ShanlanLi — APPROVED** [review]

File: docs/ingest/ingest-from-iceberg.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+| s3.region       | Optional. The region where the S3 bucket is hosted. Either `s3.endpoint` or `s3.region` must be specified.|
+| s3.access.key   | Required. Access key of the S3 compatible object store.|
+| s3.secret.key   | Required. Secret key of the S3 compatible object store.|
+| database.name   | Required. Name of the database that you want to ingest data from.|
+| table.name      | Required. Name of the table that you want to ingest data from.|
+| catalog.name    | Conditional. The name of the Iceberg catalog. It can be omitted for storage catalog but required for other catalogs.|
+| catalog.type    | Optional. The catalog type used in this table. Currently, the supported values are `storage`, `rest`, `hive` and `jdbc`. If not specified, `storage` is used. For details, see [Catalogs](#catalogs).|
+| warehouse.path  | Conditional. The path of the Iceberg warehouse. Currently, only S3-compatible object storage systems, such as AWS S3 and MinIO, are supported. It's required if the `catalog.type` is not `rest`.|
+| catalog.uri     | Conditional. The URL of the catalog. It is required when `catalog.type` is `rest`. |
```
Comment (liurenjie1024) [inline docs/ingest/ingest-from-iceberg.md]: Not only rest, all catalogs except storage needs this.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**